mayastor/src/grpc/v1/pool.rs

```diff
@@ -0,0 +1,311 @@
+            warn!("{}: gRPC method timed out, args: {}", c.id, c.args);
+        }
+
+        let fut = AssertUnwindSafe(f).catch_unwind();
+        let r = fut.await;
+
+        // Request completed, remove the marker.
+        let ctx = context_guard.take().expect("gRPC context disappeared");
+
+        match r {
+            Ok(r) => r,
+            Err(_e) => {
+                warn!("{}: gRPC method panicked, args: {}", ctx.id, ctx.args);
+                Err(Status::cancelled(format!(
+                    "{}: gRPC method panicked",
+                    ctx.id
+                )))
+            }
+        }
+    }
+}
+
+impl TryFrom<CreatePoolRequest> for PoolArgs {
+    type Error = LvsError;
+    fn try_from(args: CreatePoolRequest) -> Result<Self, Self::Error> {
+        if args.disks.is_empty() {
+            return Err(LvsError::Invalid {
+                source: Errno::EINVAL,
+                msg: "invalid argument, missing devices".to_string(),
+            });
+        }
+        Ok(Self {
+            name: args.name,
+            disks: args.disks,
+            uuid: None,
+        })
+    }
+}
+
+impl TryFrom<ImportPoolRequest> for PoolArgs {
+    type Error = LvsError;
+    fn try_from(args: ImportPoolRequest) -> Result<Self, Self::Error> {
+        if args.disks.is_empty() {
+            return Err(LvsError::Invalid {
+                source: Errno::EINVAL,
+                msg: "invalid argument, missing devices".to_string(),
+            });
+        }
+        let uuid = if let Some(ImportUuid::Uuid(uuid)) = args.uuid_value {
+            Some(uuid)
+        } else {
+            None
+        };
+        Ok(Self {
+            name: args.name,
+            disks: args.disks,
+            uuid,
+        })
+    }
+}
+
+impl PoolSvc {
+    pub fn new(interval: Duration) -> Self {
+        Self {
+            name: String::from("CSISvc"),
+            interval,
+            client_context: tokio::sync::Mutex::new(None),
+        }
+    }
+}
+
+impl From<Lvs> for Pool {
+    fn from(l: Lvs) -> Self {
+        Self {
+            uuid: l.uuid(),
+            name: l.name().into(),
+            disks: vec![l.base_bdev().bdev_uri().unwrap_or_else(|| "".into())],
+            state: PoolState::PoolOnline.into(),
+            capacity: l.capacity(),
+            used: l.used(),
+            pooltype: PoolType::Lvs as i32,
+        }
+    }
+}
+
+#[tonic::async_trait]
+impl PoolRpc for PoolSvc {
+    #[named]
+    async fn create_pool(
+        &self,
+        request: Request<CreatePoolRequest>,
+    ) -> GrpcResult<Pool> {
+        self.locked(
+            GrpcClientContext::new(&request, function_name!()),
+            async move {
+                let args = request.into_inner();
+                match PoolBackend::try_from(args.pooltype)? {
+                    PoolBackend::Lvs => {
+                        let rx = rpc_submit::<_, _, LvsError>(async move {
+                            let pool = Lvs::create_or_import(
```

Comment:
since we now have import, do we still need create_or_import, or could we just have a create?

Reply:
We will wait until we make appropriate changes in the control plane. Once we start making use of import there I will remove the import part from create. That was my thought.

Reply:
hmm yeah could do that, but we could just start with the create ("only") as well since the control-plane will not make use of the v1 yet, so when we start making use of it we can also start to use the separate create/import logic.

Reply:
The lvs function create_or_import is common between both current and v1 rpcs. I did not want to break the existing one and same time didn't want to write a entire new function. This will also help us test run v1 with the control plane when we finally start replacing the rpcs. Will refactor this slowly after that.